internal/pkg/runtime/launcher/oci/mounts_linux.go

```diff
@@ -576,10 +581,18 @@ func (l *Launcher) addUserBindMounts(mounts *[]specs.Mount) error {
 	}
 
 	for _, b := range binds {
-		if !l.singularityConf.UserBindControl {
-			sylog.Warningf("Ignoring bind mount request: user bind control disabled by system administrator")
-			return nil
+		// Special Case - user is manually requesting all of /dev to be bound
+		if b.Source == "/dev" {
+			if b.Destination != "/dev" {
+				return fmt.Errorf("host /dev may only be bound to /dev in container")
+			}
+			// Handle as we do in --no-compat, including devpts setup.
+			if err := l.addDevBind(mounts); err != nil {
```

Comment:
Should we be worried about circumstances (e.g. when --no-compat is passed) where addDevBind() is called twice - once from here, and once from addDevMounts()? Or do we want two mounts to be generated under that circumstance?

Reply:
Hmmmm... I'm inclined to leave it as-is unless it causes a problem.

You can, for instance do someting like have `--no-compat` bind mount `/dev`.... a sysadmin has set `SINGULARITY_BINDS` to mount also a `/dev/mydevice` ... but you want to mask that by bind mounting `/dev` again over what's there.

Unlikely... probably... but I don't think allowing that is incorrect.